tests/conftest.py

```diff
@@ -149,103 +179,228 @@ def test_something(remote: Remote):
     ```
     """
 
-    slurm_cluster_hostname = request.param
-
-    if not slurm_cluster_hostname:
+    cluster_name = request.param
+    if not cluster_name:
         pytest.skip("Requires ssh access to a SLURM cluster.")
 
-    # TODO: Re-enable this, but only on tests that say that they run jobs on the
-    # cluster.
-    # with cancel_all_milatools_jobs_before_and_after_tests(slurm_cluster_hostname):
-    return slurm_cluster_hostname
+    clusters_in_maintenance = os.environ.get("CLUSTER_DOWN", "").split(",")
+    if cluster_name in clusters_in_maintenance:
+        pytest.skip(reason=f"Cluster {cluster_name} is down for maintenance.")
```

Comment:
Removed in cc4f29b1be9a19237fc810db5342768b4ae00b75.
The tests are already split by cluster, so I think we have what you want already. This was more meant for local testing, before the `SLURM_CLUSTER` env variable and the `cluster` fixture were used for tests, some tests (ex. `test_code_command.py::test_check_disk_quota`) would be parametrized for each cluster manually, and would test each cluster one by one. This would more easily allow a cluster to be skipped. This was bad, thanks for pointing that out! :)

Reply:
Indeed we have but what I mean is to have one integration tests suit by cluster to shown separately here :
![image](https://github.com/mila-iqia/milatools/assets/6757005/067ca184-5b98-48a9-aed8-a34109ea9ad1)

BTW, I don't we should need to test multiple versions of python in the integration tests? Python should not affect anything when executing the commands on the different clusters for example. What changes is the cluster right?

Reply:
Yes that's actually already implemented! (For example `full_cluster_tests.yaml` runs this for all clusters, but it's currently only run periodically and on pushes to `master`).

Here in the `build.yaml` workflow I'm only doing the for-loop with the mila cluster.

About the second part, I'm not sure if you're saying that we should or shouldn't use different versions of Python, but at the moment I'm only using one version of Python for integration tests. Is that good or bad? :sweat_smile:

Reply:
I'd only use the oldest and newest Pythons supported, on the assumption that this should be sufficient to catch attempts at using too-new features, and any compatibility breaks in the newest Python release.

Reply:
Oh yes my bad I missed a few things here!